### PR TITLE
Add `docker cp` and `docker exec`

### PR DESCRIPTION
**Pull request by joshuatcasey**
Add helpers for `docker cp`, `docker exec`, and "docker exec with bash".

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ type Docker struct {
 	}
 
 	Pull DockerPull
+	Copy DockerCopy
+	Exec DockerExec
```
Comment (ryanmoran) [inline docker.go:33]: Can we namespace these under the `Container` subgroup? Both `cp` and `exec` operate on a container that is already running, and the `docker` command also has them nested under its own `container` subcommand (I believe they exist at the top-level for backwards compatibility).

Comment (joshuatcasey) [reply docker.go:33]: Good find about the subcommands. I'll move them.

Comment (joshuatcasey) [reply docker.go]: I've also put them in at the correct alphabetical order since that seems preferred

File: docker.go [hunk 2]
```diff
@@ -374,6 +380,47 @@ func (s DockerContainerStop) Execute(containerID string) error {
...
 }
 
+type DockerContainerCopy struct {
+	executable Executable
+}
+
+func (docker DockerContainerCopy) Execute(source, dest string) error {
+	stderr := bytes.NewBuffer(nil)
+	err := docker.executable.Execute(pexec.Execution{
+		Args:   []string{"cp", source, dest},
```
Comment (ryanmoran) [inline docker.go:390]:
```suggestion
		Args:   []string{"container", "cp", source, dest},
```

File: docker.go [hunk 3]
```diff
@@ -374,6 +380,47 @@ func (s DockerContainerStop) Execute(containerID string) error {
...
+}
+
+type DockerContainerExec struct {
+	executable Executable
+}
+
+func (docker DockerContainerExec) Execute(container string, args ...string) error {
+	stderr := bytes.NewBuffer(nil)
+	execution := pexec.Execution{
+		Args:   []string{"exec", container},
```
Comment (ryanmoran) [inline docker.go:407]:
```suggestion
		Args:   []string{"container", "exec", container},
```

**Review by ryanmoran — CHANGES_REQUESTED** [review]
Now that we've nested these commands under the `Container` group, can we align the actual commands invoked to match their Go API?

File: docker.go (hunk 2 above)
Comment (joshuatcasey) [reply docker.go]: done with new commit

File: docker.go (hunk 3 above)
Comment (joshuatcasey) [reply docker.go]: done with new commit

**Review by ryanmoran — APPROVED** [review]

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**